feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+
+        # Solve
+        model.verbose = 0
+        model.optimize()
+
+        # Solution
+        selected = [i for i in data.X if float(x[i].x) >= 0.99]
+
+        if self.verbose:
+            print("=" * 40)
+            print("MAX COVER OBJECTIVE:", model.objective_value)
+            print("NUM ROWS COVERED:", num_row_covered.x,
+                  "coverage: {:.2f}".format(num_row_covered.x / data.matrix.shape[0]))
+            print("SIZE:", len(selected),
+                  "reduction: {:.2f}".format((data.matrix.shape[1] - len(selected)) / data.matrix.shape[1]))
+            print("SELECTED:", selected)
+            print("STATUS:", model.status)
+            print("=" * 40)
+
+        assert model.status == OptimizationStatus.OPTIMAL
+
+        # Return
+        return selected
+
+    @staticmethod
+    def _validate_args(selection_size):
+        if selection_size is not None:
+            if selection_size <= 0:
+                raise ValueError("Selection size must be greater than zero.")
+
+
+# TODO not sure if this should use _BaseDispatcher or not. We can decide later
+class _TextBased(_BaseSupervisedSelector):
+    """
+    featrization and optimization are done in 'fit' function.
+    The 'transform' function is used to show the selected contents
+    """
+
+    def __init__(self, seed: int, num_features: Num,
+                 featurization_method: TextWiser, optimization_method: str, cost_metric: str,
+                 trials: int = 10):
+        # Call constructor of parent class _BaseSupervisedSelector
+        super().__init__(seed)
+
+        self.selected_features_ = None
+        self.num_features = num_features
+        self.featurization_method = featurization_method
+        self.selection_size = num_features
+        self.optimization_method = optimization_method
+        self.cost_metric = cost_metric
+        self.trials = trials
+        self.content_selector = ContentSelector(selection_size=num_features,
+                                                seed=self.seed, trials=trials, verbose=True)
+
+    def fit(self, data: pd.DataFrame, labels: pd.DataFrame) -> NoReturn:
+
+        # Call fit method from parent class _BaseSupervisedSelector
+        super().fit(data, labels)
+
+        # Get the text columns dynamically
+        text_columns = [col for col in data.columns if col.startswith("item")]
+
+        selected_indicies = self.content_selector.run_content_selection(data, labels, self.featurization_method,
+                                                                        self.selection_size,
+                                                                        self.optimization_method, self.cost_metric)
+
+        # Only select the features that were selected during fit
+        selected_features = [col for i, col in enumerate(text_columns) if i in selected_indicies]
+        self.selected_features_ = selected_features
```

Comment:
So we are adding a new field called selected features in the init() and then use it here.. which makes total sense

This code is fine as-is but within this library.. but to be consistent with every other method.. maybe we can drop this variable and instead use the abs_score field from the base..

In the public api, we have a get_abs_score() method. We need to come up with some "score" to return for each feature.. For instnance, a linear selection can return regression coefficients as the "abs_score". 

I don't see any clear definition of a "score" for TextBased. We can treat score as 0/1 binary, set it to score=1 if selected and 0 otherwise. Setting abs_score to 1 using selected_indices this way replaces the selected_features variable here

We can also set self.num_features = len(selected_indices) (IN CASE, we end up selecting less features/items than asked)

Then in transform() we can make a call to get_top_k() like in other methods, and it returns top_num_features from abs_scores --which should return the selected indices without tracking a separate "selected_features" 

As a bonus, the get_abs_score() method in the public API also works for TextBased.

Reply:
Great Idea! 
Made following changes:
Instead of using the selected_features variables to store the selected columns, I set the absolute score for each feature in the fit() method and storr the abs_scores field. In transform() method, the top_k selected features will return when the num_features = len(selected_indicies)